mlrun/api/utils/singletons/db.py

```diff
@@ -14,8 +14,11 @@ def get_db() -> DBInterface:
     return db
 
 
-def initialize_db():
+def initialize_db(override_db=None):
     global db
+    if override_db:
+        db = override_db
+        return
```

Comment:
It is a singleton, but this function is `initialize_db` - which means is creates the singleton instance, and for testing purposes we want to give it the instance which will be the singleton instead of it creating one

Reply:
I understand, that behavior puts it outside of the realm of singleton, though.
I guess wouldn't be resolved here but sounds like it's not really a singleton anymore